File: src/main/java/net/hectus/neobb/turn/default_game/other/TBoat.java
package net.hectus.neobb.turn.default_game.other;

import net.hectus.neobb.NeoBB;
import net.hectus.neobb.player.NeoPlayer;
import net.hectus.neobb.turn.default_game.attributes.clazz.WaterClazz;
import net.hectus.neobb.turn.default_game.attributes.function.EventFunction;
import net.hectus.neobb.util.Modifiers;
import org.bukkit.Bukkit;
import org.bukkit.Material;
import org.bukkit.entity.Boat;
import org.bukkit.inventory.ItemStack;

public class TBoat extends OtherTurn<Boat> implements EventFunction, WaterClazz {
    public TBoat(NeoPlayer player) { super(player); }
    public TBoat(Boat data, NeoPlayer player) { super(data, data.getLocation(), player); }

    @Override
    public boolean requiresUsageGuide() { return true; }

    @Override
    public ItemStack item() {
        return new ItemStack(Material.OAK_BOAT);
    }

    @Override
    public int cost() {
        return 3;
    }

    @Override
    public void triggerEvent() {
        player.nextPlayer().addModifier(Modifiers.P_DEFAULT_BOAT_DAMAGE);
        Bukkit.getScheduler().runTaskTimer(NeoBB.PLUGIN, r -> {
            if (player.nextPlayer().hasModifier(Modifiers.P_DEFAULT_BOAT_DAMAGE)) {
                player.nextPlayer().player.damage(1.0);
            } else {
                r.cancel();
            }
        }, 0, 20);
    }
}

File: src/main/kotlin/net/hectus/neobb/turn/default_game/other/TBoat.kt
package net.hectus.neobb.turn.default_game.other

import com.marcpg.libpg.storing.Cord
import net.hectus.neobb.player.NeoPlayer
import net.hectus.neobb.turn.default_game.attribute.clazz.WaterClazz
import net.hectus.neobb.turn.default_game.attribute.function.EventFunction
import net.hectus.neobb.util.Modifiers
import net.hectus.neobb.util.bukkitRunTimer
import org.bukkit.Material
import org.bukkit.entity.Boat
import org.bukkit.inventory.ItemStack

class TBoat(data: Boat?, cord: Cord?, player: NeoPlayer?) : OtherTurn<Boat>(data, cord, player), EventFunction, WaterClazz {
    override val cost: Int = 3

    override fun item(): ItemStack = ItemStack(Material.OAK_BOAT)

    override fun triggerEvent() {
        player!!.nextPlayer().addModifier(Modifiers.Player.Default.BOAT_DAMAGE)
        bukkitRunTimer(0, 20) { r ->
            if (player.nextPlayer().hasModifier(Modifiers.Player.Default.BOAT_DAMAGE)) {
                player.nextPlayer().damage(1.0)
            } else {
                r.cancel()
            }
        }
    }
}
